lib/datadog/core/configuration.rb

```diff
@@ -83,17 +80,22 @@ def configuration
       # The yielded configuration `c` comes pre-populated from environment variables, if
       # any are applicable.
       #
-      # @param [Datadog::Core::Configuration::Settings] configuration the base configuration object. Provide a custom
-      #   instance if you are managing the configuration yourself. By default, the global configuration object is used.
       # @yieldparam [Datadog::Core::Configuration::Settings] c the mutable configuration object
-      def configure(configuration = self.configuration)
-        # Wrap block with global option validation
-        wrapped_block = proc do |c|
-          yield(ValidationProxy::Global.new(c))
+      def configure
```

Comment:
Not sure if this got reintroduced in 1.0, but the `configuration = self.configuration` has no valid public use (no valid internal use today either).

We used to allow an argument to be passed to `#configure`, but that use case has moved to `#configure_onto`.

Reply:
In theory should we permit a `Settings` object to be given to it? If not, or there's no real case for this, we can remove.